Create new file

vasubhardwaj011 opened the pull request:
- add single space in comment line. [ed960df](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/ed960df763d6b1b005663fc4c50b8a767e5146d6)
- write code to remove last occurrence of character of the string. [6b7c55e](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/6b7c55e80c19778f8aca4a5aaccbda2fd40e1380)
- write code to remove all occurrence of character of the string. [8e4041c](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/8e4041cc10949121183b0cde80b8d738656afabb)
- updated 23_remove_last_occurrence_of_character.js. [4ed0850](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/4ed08502072cb001886bdbf865c311de50f6cbca)
- updated 24_remove_all_occurrence_of_character.js. [11d7b9e](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/11d7b9e8546388c0884779b2f58f564388092cb0)
- update code to avoid inbuilt functions and refactor logic. [cad124d](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/cad124dc773a4196b52e583c21b406c1c6a0b15f)
- update code to avoid inbuilt functions and refactor logic. [f9d0e5a](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/f9d0e5a2200d43f06fe571213f3e9f51fe69e201)]
- add JavaScript program to remove all repeated characters from a string. [759a277](https://github.com/vasubhardwaj011/javaScript_questions/pull/1/commits/759a277e4eaf8495854345af216f273e79df2a62)]

vasubhardwaj011 committed:
add single space in comment line

vasubhardwaj011 committed:
feat: 23_remove_last_occurrence_of_character.js

vasubhardwaj011 committed:
feat: 24_remove_all_occurrence_of_character.js

Rohit2820 commented on 24_remove_all_occurrence_of_character.js:
```diff
@@ -0,0 +1,25 @@
+
+    if (text.string[i] == text.search) {
+        text.count++;
+        if (text.count == text.count) {
```
stop drinking in office hours

Rohit2820 commented on 23_remove_last_occurrence_of_character.js:
```diff
@@ -0,0 +1,32 @@
+let text = {
+    string: "My Name is Vasu",
+    search: "a",
+    count: 0
```
what is the use of this

Rohit2820 commented on 23_remove_last_occurrence_of_character.js:
```diff
@@ -0,0 +1,32 @@
+
+for (let i = 0; i < text.string.length; i++) {
+
+    if (text.string[i] == text.search) {
```
Told you already, dont drink and code

vasubhardwaj011 replied:
I have understood your suggestion and updated this code.
Thank you

vasubhardwaj011 committed:
feat: update 23_remove_last_occurrence_of_character.js

vasubhardwaj011 reviewed (commented):
updated 23_remove_last_occurrence_of_character.js
please approve it.

vasubhardwaj011 committed:
feat: update 24_remove_all_occurrence_of_character.js

kapilPerlStreet commented on 24_remove_all_occurrence_of_character.js:
```diff
@@ -0,0 +1,20 @@
+
+    if (text.string[i] == text.search) {
+
+        text.string = text.string.replace(text.string[i], text.replace_with);
```
@vasubhardwaj011 

Please do not use in-built functions like `replace`.

vasubhardwaj011 replied:
okk

kapilPerlStreet commented on 23_remove_last_occurrence_of_character.js:
```diff
@@ -0,0 +1,28 @@
+//          \ /
+//           V
+
+text.string = text.string.slice(0, text.last_occurrence_char_index) + text.string.slice((text.last_occurrence_char_index + 1), text.string.length);
```
Please do not use in-built functions.

vasubhardwaj011 replied:
okk, I will change it.
Thank you for your review.

vasubhardwaj011 committed:
refactor: update code to avoid inbuilt functions and refactor logic

- Replaced previous implementation that used slice method with a loop
- Implemented a new approach to construct the string without using the slice method

vasubhardwaj011 committed:
refactor: update code to avoid inbuilt functions and refactor logic

- Replaced previous implementation that used replace method
- Implemented a new approach to construct the string without using the replace method

vasubhardwaj011 committed:
feat: add JavaScript program to remove all repeated characters from a string

- Implemented a program to remove repeated occurrences of a specified character from the string
- Added logic to count occurrences of the search character and construct a new string without repeated characters
